### ISSUE TITLE
Replace the Mono font with something a little nicer

### ISSUE THREAD

**Issue by jmaris** [InfiniTimeOrg/InfiniTime#955]
### Verification

- [X] I searched for similar feature request and found none was relevant.

### Pitch us your idea!

PinetimeLite has a Nice Sans Serif font, could we do the same?

**Comment by JF002** [comment]
Do you have a link to such font (preferably one with a license that allow us to use it in an open source project) ?

**Comment by ainola** [comment]
There are many such compatible fonts! :)

Perhaps this list would help: https://wiki.archlinux.org/title/Fonts#Latin_script

You can't go wrong with the  Liberation family. Cantarell is used by gnome. Noto is a fan favorite. Lato is a personal one.

**Comment by geekbozu** [comment]
This would be a good test for the filesystem as well. I wonder how painful loading a font from the FS would be. If its not exceptionally slow it might be worth writing it to external flash in the next update so we can strip it from the rom in the future!

**Comment by Riksu9000** [comment]
I did some testing. I found that the optimal thickness for a non antialiased font is 2 pixels. The current font is about 2-3 pixels and a font with a thickness of a single pixel is too thin to use in some cases. Ideally we would have multiple weights, or at least a regular and a monospaced font. Unfortunately, unless I made a mistake, it seems that a non monospaced font uses ~3700 more bytes due to kerning I suppose.

This is Roboto Medium. These screenshots are a bit blurry, but in reality this looks very nice and sharp. Here we can see another issue however. The font conversion isn't pixel perfect. If we look at the word InfiniTime, we can see that there is a larger gap between i and T than T and i. There are many more cases like this where the spacing isn't as expected.

We need to find a font that is the optimal thickness, preferrably with multiple weights, and looks good at very small sizes. I don't know if the spacing issue is due to the font or the LVGL converter, but it should be taken into consideration.

![license](https://user-images.githubusercontent.com/37774658/151662447-94e35078-6570-421a-95ec-d8e6255c8d7c.jpg)
![settings](https://user-images.githubusercontent.com/37774658/151662448-948dafea-599f-46c4-be8c-20fb25d1bb7b.jpg)
![watchface](https://user-images.githubusercontent.com/37774658/151662450-d0685e55-c12e-497c-9658-4cc1bf77e59b.jpg)

**Comment by ainola** [comment]
Size was something I was a little concerned about. These look super good though! Much more readable.

**Comment by JF002** [comment]
Yes, I think it looks good, but as you say, the spacing looks inconsistent. The word "format" on the 2nd screenshot look like "forma t" with a space before the 't'.

**Comment by medeyko** [comment]
I think that Jetbrains mono font is more stylish. I'd prefer it not to be removed.

Also variable width font should not be used on watch faces and in most apps, to ensure that the position of characters is constant. It is especially important for dynamically changing texts.

Also I believe that variable-length font inevitable requires more hardware resources, both memory and CPU.

**Comment by ainola** [comment]
On 2022-02-06 22:05, medeyko wrote:
>Also I believe that variable-length font inevitable requires more hardware resources, both memory and CPU.

I'd be interested in getting some benchmarks because this makes me think 
of the "SSL takes more hardware resources" that may have been valid on 
older hardware but perhaps is not the case nowadays.

But that is true that it would cause more maintenance to ensure proper 
kerning, etc.

**Comment by jarinox** [comment]
> Yes, I think it looks good, but as you say, the spacing looks inconsistent. The word "format" on the 2nd screenshot look like "forma t" with a space before the 't'.

Even with this inconsistency I think the proposed font looks much nicer than the current monospace font.
